contents/node/nodejs.md

```diff
@@ -0,0 +1,208 @@
+# Introduction to Node.js
+
+Author: Rachael Sim
+
+# Overview
+
+_This chapter assumes that the reader is familiar with JavaScript and asynchronous programming. If you are not familiar with asynchronous programming, a good resource to checkout is the [asynchronous programming section of the You Don't Know JS guide ](https://github.com/getify/You-Dont-Know-JS/blob/master/async%20&%20performance/README.md#you-dont-know-js-async--performance) as asynchronous programming is key in Node._
+
+*Node.js* is a JavaScript runtime built on Chrome’s V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient.
+
+# Benefits
+
+Why should you use Node?
+
+## Benefit 1: Node.js is single-threaded
+
+Node is designed to be **event-driven**. When an event happens e.g. an I/O event is complete, the event handler/callback function will be enqueued in a queue where it will be subsequently scheduled to run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c). The main event loop (where Javascript code is executed) is single-threaded.
+
+This means that we can avoid thread overheads and synchronization problems including deadlocks and race conditions.
+
+## Benefit 2: Node.js is fast for I/O intensive tasks
```

Comment:
we don't use Node.js in the heading of other benefits. Remove it here as well, for consistency?